### PR TITLE
Add LMove command for db & server

### PR DESCRIPTION
**Pull request by Roderland**
[LMove](https://redis.io/commands/lmove/) command in redis.

### CODE REVIEW

File: cmd/command.go [hunk 1]
```diff
@@ -223,6 +223,39 @@ func rpop(cli *Client, args [][]byte) (interface{}, error) {
 	return popInternal(cli.db, args, false)
 }
 
+func lmove(cli *Client, args [][]byte) (interface{}, error) {
```
Comment (gozeloglu) [inline cmd/command.go:226]: I think, `lMove` is a more convenient naming for the Go language. You know Go follows the camelCase naming convention. 
```suggestion
func lMove(cli *Client, args [][]byte) (interface{}, error) {
```

File: list.go [hunk 2]
```diff
@@ -96,6 +96,31 @@ func (db *RoseDB) RPop(key []byte) ([]byte, error) {
 	return db.popInternal(key, false)
 }
 
+// LMove Atomically returns and removes the first/last element of the list stored at source,
```
Comment (gozeloglu) [inline list.go:99]:
```suggestion
// LMove atomically returns and removes the first/last element of the list stored at source,
```

File: cmd/command.go [hunk 3]
```diff
@@ -223,6 +223,39 @@ func rpop(cli *Client, args [][]byte) (interface{}, error) {
...
+		return nil, errSyntax
+	}
+
+	value, err := cli.db.LMove(srcKey, dstKey, srcIsLeft, dstIsLeft)
+	if err != nil {
+		return nil, err
+	}
+
+	return value, nil
+}
```
Comment (gozeloglu) [inline cmd/command.go:257]: I think you can return directly the function call. 
```suggestion
	return cli.db.LMove(srcKey, dstKey, srcIsLeft, dstIsLeft)
}
```

File: cmd/command.go (hunk 1 above)
Comment (Roderland) [reply cmd/command.go:226]: @gozeloglu 
Yes, "lMove" is better, but for consistency with other existing command function names. Maybe we can update all names in next pr.
https://github.com/flower-corp/rosedb/blob/7b155b6de9816eaad04913c58206dfecd8444f65/cmd/cli.go#L20-L25

Comment (gozeloglu) [reply cmd/command.go:226]: No worry about that. I've changed the names in my PR: https://github.com/flower-corp/rosedb/pull/152. It is still waiting for merge.

Comment (Roderland) [reply cmd/command.go]: @gozeloglu OK

**Comment by roseduan** [comment]
> @roseduan Please have a look at this PR.

OK, thanks for you review.

**Pull request merged**